benchmarks/copy_view_performance/copy_view_performance.cpp

```diff
@@ -0,0 +1,306 @@
+        double my_value = tmp_view_collection.sum_views(i, execute_kernel);
+        if (my_value > valueToUpdate) valueToUpdate = my_value;
+      },
+      Kokkos::Max<double>(max_value));
+  double view_collection_time = view_collection_timer.seconds();
+
+  bool success = std::fabs(max_value - N * NVIEWS) < 1.E-6;
+  std::cout << "View Time = " << view_collection_time << " seconds"
+            << std::endl;
+  if (success) {
+    std::cout << "SUCCESS" << std::endl;
+  } else {
+    std::cout << "FAILURE" << std::endl;
+  }
+}
+
+void test_view_collection_serial(int N, int num_iter, bool execute_kernel) {
+  ViewCollection view_collection(N);
+
+  Kokkos::Timer view_collection_timer;
+  double max_value = 0.0;
+  // Max Reduction boilerplate code taken from slide 53 of
+  // kokkos-tutorials/LectureSeries/KokkosTutorial_02_ViewsAndSpaces.pdf
+  for (int i = 0; i < num_iter; ++i) {
+    // NOTE: The following lines expose the Kokkos View overheads
+    ViewCollection tmp_view_collection = view_collection;
+    double my_value = tmp_view_collection.sum_views(i, execute_kernel);
+    if (my_value > max_value) max_value = my_value;
+  }
+  double view_collection_time = view_collection_timer.seconds();
+
+  bool success = std::fabs(max_value - N * NVIEWS) < 1.E-6;
+  std::cout << "View Time 2 = " << view_collection_time << " seconds"
+            << std::endl;
+  if (success) {
+    std::cout << "SUCCESS" << std::endl;
+  } else {
+    std::cout << "FAILURE" << std::endl;
+  }
+}
+
+// Based on kokkos-tutorials/Exercises/02/Solution/exercise_2_solution.cpp
+int main(int argc, char* argv[]) {
+  int N               = 1;
+  int num_iter        = 1 << 27;
+  bool execute_kernel = false;
+
+  for (int i = 0; i < argc; i++) {
+    if ((strcmp(argv[i], "-N") == 0)) {
+      N = atoi(argv[++i]);
+      if (N < 1) {
+        std::cout << "Array extent must be >= 1" << std::endl;
+        exit(1);
+      }
+    } else if (strcmp(argv[i], "-i") == 0) {
+      num_iter = atoi(argv[++i]);
+      if (num_iter < 1) {
+        std::cout << "Number of iterations must be >= 1" << std::endl;
+        exit(1);
+      }
+    } else if (strcmp(argv[i], "-k") == 0) {
+      execute_kernel = true;
+    } else if ((strcmp(argv[i], "-h") == 0)) {
+      printf("  Options:\n");
+      printf("  -N <int>: Array extent\n");
+      printf("  -i <int>: Number of iterations\n");
+      printf("  -k:       Execute the summation kernel\n");
+      printf("  -h:       Print this message\n\n");
+      exit(1);
+    }
+  }
+
+  std::cout << "Array extent = " << N << std::endl;
+  std::cout << "Iterations = " << num_iter << std::endl;
+  std::cout << "Execute summation kernel = " << std::boolalpha << execute_kernel
+            << std::noboolalpha << std::endl;
+
+  // Test inside a Kokkos kernel.
+  Kokkos::initialize(argc, argv);
+  { test_view_collection_kk(N, num_iter, execute_kernel); }
+
+  // Test outside Kokkos kernel.
```

Comment:
Here, do you mean test on host?  Would it make sense here to exercise control flow through verifying if on host or device , vs. a re-named test of duplicate code (see comment above)?